src/main/scala/nl.knaw.dans.easy.solr4files/components/DDM.scala

```diff
@@ -0,0 +1,114 @@
+
+import scala.xml.{ Node, XML }
+
+class DDM(xml: Node) extends DebugEnhancedLogging {
+
+  val accessRights: String = (xml \ "profile" \ "accessRights").text
+  val solrLiterals: SolrLiterals =
+    (xml \ "dcmiMetadata" \ "identifier").filter(isDOI).map(n => ("dataset_doi", n.text)) ++
+      (xml \ "profile" \ "creatorDetails").map(n => ("dataset_creator", n.text)) ++
+      (xml \ "profile" \ "creator").map(n => ("dataset_creator", n.text)) ++
+      (xml \ "profile" \ "title").map(n => ("dataset_title", n.text)) ++
+      (xml \ "profile" \ "audience").flatMap(n => Seq(
+        ("dataset_narcis_audience", n.text), // TODO https://github.com/DANS-KNAW/easy-dtap/pull/131#pullrequestreview-60582037
+        ("dataset_audience", audienceMap.getOrElse(n.text, ""))
+      )) ++
+      (xml \ "dcmiMetadata" \ "subject").flatMap { n =>
+        val abrMap = getAbrMap(n) // FIXME
+        if (abrMap.isEmpty) Seq(("dataset_subject", n.text))
+        else {
+          Seq(
+            ("dataset_abr_subject", n.text), // TODO https://github.com/DANS-KNAW/easy-dtap/pull/131#pullrequestreview-60582037
+            ("dataset_subject", abrMap.get.getOrElse(n.text, ""))
+          )
+        }
+      } ++
+      (xml \ "dcmiMetadata" \ "relation").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "conformsTo").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "isVersionOf").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "hasVersion").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "isReplacedBy").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "replaces").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "isRequiredBy").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "requires").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "isPartOf").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "hasPart").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "isReferencedBy").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "references").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "isFormatOf").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "hasFormat").map(n => ("dataset_relation", n.text))
+  (xml \ "dcmiMetadata" \ "temporal").map(n => ("dataset_coverage", n.text))
+  // TODO spatial https://github.com/DANS-KNAW/easy-schema/blob/acb6506/src/main/assembly/dist/docs/examples/ddm/example2.xml#L280-L320
+  // TODO   "dataset_coverage" -> (xml \ "dcmiMetadata" \ "spatial").text,
+}
+
+object DDM {
+
+  private val xsiURI = "http://www.w3.org/2001/XMLSchema-instance"
+
+  private val abrPrefix = "abr:ABR"
+
+  private def getAbrMap(ddmSubjectNode: Node): Option[Map[String, String]] = {
+    ddmSubjectNode
+      .attribute(xsiURI, "type")
+      .map(_.text)
+      .filter(_.startsWith(abrPrefix))
+      .flatMap(abrMaps.get)
```

Comment:
do you really need this `get` here. That's unsafe!!!

Reply:
`get` returns an option which I flatMap with the option returned by `attribute`, so what is unsafe?